tests/test_pruner.cpp

```diff
@@ -298,116 +303,234 @@ template <class FT> int test_auto_prune(size_t n)
             PRUNER_METRIC_PROBABILITY_OF_SHORTEST, true);
   status |= !(pruning.expectation <= 1.0);
   cerr << "Probability " << pruning.expectation << endl;
+  cost = 0.;
+  for (size_t i = 0; i < 2 * n; ++i)
+  {
+    cerr << pruning.detailed_cost[i] << "\t";
+    cost += pruning.detailed_cost[i];
+  }
+  cerr << endl << "Total Cost " << cost << endl;
+
   status |= !(pruning.expectation > 0.0);
   status |= !(pruning.radius_factor >= 1.0);
   status |= !(pruning.coefficients[0] == 1.0);
 
+  if (status)
+    exit(1);
```

Comment:
My preference would be to print an error if a test failed and to continue. A bit more boilerplate code but it would give the maximum amount of feedback.